code/Model/Recipient/EmailRecipient.php

```diff
@@ -229,15 +229,7 @@ protected function getFormParent()
             $formClass = $this->FormClass;
             return $formClass::get()->byID($this->FormID);
         }
-
-        // Revert to checking for a form from the session
-        // LeftAndMain::sessionNamespace is protected.
-        $sessionNamespace = $this->config()->get('session_namespace') ?: CMSMain::class;
-
-        $formID = Controller::curr()->getRequest()->getSession()->get($sessionNamespace . '.currentPage');
-        if ($formID) {
-            return UserDefinedForm::get()->byID($formID);
-        }
```

Comment:
Storing the current record ID in the session was deprecated, and is being removed in https://github.com/silverstripe/silverstripe-admin/pull/1867

From what I can tell, the relation is always set anyway so we shouldn't ever get to this.